[2023-03-30T20:00:04.089-0400] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.run_producer scheduled__2023-03-30T00:00:00+00:00 [queued]>
[2023-03-30T20:00:04.099-0400] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.run_producer scheduled__2023-03-30T00:00:00+00:00 [queued]>
[2023-03-30T20:00:04.099-0400] {taskinstance.py:1282} INFO - 
--------------------------------------------------------------------------------
[2023-03-30T20:00:04.099-0400] {taskinstance.py:1283} INFO - Starting attempt 1 of 2
[2023-03-30T20:00:04.099-0400] {taskinstance.py:1284} INFO - 
--------------------------------------------------------------------------------
[2023-03-30T20:00:04.109-0400] {taskinstance.py:1303} INFO - Executing <Task(BashOperator): run_producer> on 2023-03-30 00:00:00+00:00
[2023-03-30T20:00:04.117-0400] {standard_task_runner.py:55} INFO - Started process 33921 to run task
[2023-03-30T20:00:04.123-0400] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'air_quality_pipeline', 'run_producer', 'scheduled__2023-03-30T00:00:00+00:00', '--job-id', '75', '--raw', '--subdir', 'DAGS_FOLDER/__pycache__/pipeline.py', '--cfg-path', '/var/folders/8z/pwbx6b3n3mnd1g1f397fv1100000gn/T/tmpt78lm49n']
[2023-03-30T20:00:04.125-0400] {standard_task_runner.py:83} INFO - Job 75: Subtask run_producer
[2023-03-30T20:00:04.171-0400] {task_command.py:388} INFO - Running <TaskInstance: air_quality_pipeline.run_producer scheduled__2023-03-30T00:00:00+00:00 [running]> on host srivardhans-air.starry-inc.net
[2023-03-30T20:00:04.216-0400] {taskinstance.py:1511} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=yourname
AIRFLOW_CTX_DAG_ID=air_quality_pipeline
AIRFLOW_CTX_TASK_ID=run_producer
AIRFLOW_CTX_EXECUTION_DATE=2023-03-30T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-03-30T00:00:00+00:00
[2023-03-30T20:00:04.218-0400] {subprocess.py:63} INFO - Tmp dir root location: 
 /var/folders/8z/pwbx6b3n3mnd1g1f397fv1100000gn/T
[2023-03-30T20:00:04.219-0400] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'python3 "/Users/srivardhan/Desktop/Masters/Masters Project/Kafka Implementation/kafka-pipeline-examples/producer-consumer/producer.py"']
[2023-03-30T20:00:04.231-0400] {subprocess.py:86} INFO - Output:
[2023-03-30T20:01:08.126-0400] {subprocess.py:93} INFO - Fetching Data from Purple Air sensors
[2023-03-30T20:01:08.130-0400] {subprocess.py:93} INFO - Traceback (most recent call last):
[2023-03-30T20:01:08.131-0400] {subprocess.py:93} INFO -   File "/Users/srivardhan/Desktop/Masters/Masters Project/Kafka Implementation/kafka-pipeline-examples/producer-consumer/producer.py", line 23, in <module>
[2023-03-30T20:01:08.132-0400] {subprocess.py:93} INFO -     producer.send('air-quality', value=data)
[2023-03-30T20:01:08.132-0400] {subprocess.py:93} INFO -   File "/Users/srivardhan/opt/anaconda3/envs/kafka-pipeline/lib/python3.7/site-packages/kafka/producer/kafka.py", line 576, in send
[2023-03-30T20:01:08.133-0400] {subprocess.py:93} INFO -     self._wait_on_metadata(topic, self.config['max_block_ms'] / 1000.0)
[2023-03-30T20:01:08.134-0400] {subprocess.py:93} INFO -   File "/Users/srivardhan/opt/anaconda3/envs/kafka-pipeline/lib/python3.7/site-packages/kafka/producer/kafka.py", line 703, in _wait_on_metadata
[2023-03-30T20:01:08.134-0400] {subprocess.py:93} INFO -     "Failed to update metadata after %.1f secs." % (max_wait,))
[2023-03-30T20:01:08.135-0400] {subprocess.py:93} INFO - kafka.errors.KafkaTimeoutError: KafkaTimeoutError: Failed to update metadata after 60.0 secs.
[2023-03-30T20:01:08.159-0400] {subprocess.py:97} INFO - Command exited with return code 1
[2023-03-30T20:01:08.171-0400] {taskinstance.py:1775} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/srivardhan/opt/anaconda3/envs/kafka-pipeline/lib/python3.7/site-packages/airflow/operators/bash.py", line 197, in execute
    f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2023-03-30T20:01:08.175-0400] {taskinstance.py:1326} INFO - Marking task as UP_FOR_RETRY. dag_id=air_quality_pipeline, task_id=run_producer, execution_date=20230330T000000, start_date=20230331T000004, end_date=20230331T000108
[2023-03-30T20:01:08.183-0400] {standard_task_runner.py:105} ERROR - Failed to execute job 75 for task run_producer (Bash command failed. The command returned a non-zero exit code 1.; 33921)
[2023-03-30T20:01:08.216-0400] {local_task_job.py:212} INFO - Task exited with return code 1
[2023-03-30T20:01:08.232-0400] {taskinstance.py:2585} INFO - 0 downstream tasks scheduled from follow-on schedule check
